packages/compass-import-export/src/modules/export.ts

```diff
@@ -0,0 +1,531 @@
+
+type FetchFieldsToExportErrorAction = {
+  type: ExportActionTypes.FetchFieldsToExportError;
+  aborted?: boolean;
+  errorMessage?: string;
+};
+
+type FetchFieldsToExportSuccessAction = {
+  type: ExportActionTypes.FetchFieldsToExportSuccess;
+  fieldsToExport: FieldsToExport;
+};
+
+type ToggleFieldToExportAction = {
+  type: ExportActionTypes.ToggleFieldToExport;
+  fieldId: string;
+};
+export const toggleFieldToExport = (fieldId: string) => ({
+  type: ExportActionTypes.ToggleFieldToExport,
+  fieldId,
+});
+
+type ToggleExportAllSelectedFieldsAction = {
+  type: ExportActionTypes.ToggleExportAllSelectedFields;
+};
+export const toggleExportAllSelectedFields = () => ({
+  type: ExportActionTypes.ToggleExportAllSelectedFields,
+});
+
+type AddFieldToExportAction = {
+  type: ExportActionTypes.AddFieldToExport;
+  path: SchemaPath;
+};
+export const addFieldToExport = (path: SchemaPath) => ({
+  type: ExportActionTypes.AddFieldToExport,
+  path,
+});
+
+type ReadyToExportAction = {
+  type: ExportActionTypes.ReadyToExport;
+  selectedFieldOption?: 'all-fields';
+};
+
+export const readyToExport = (): ReadyToExportAction => ({
+  type: ExportActionTypes.ReadyToExport,
+});
+
+type RunExportAction = {
+  type: ExportActionTypes.RunExport;
+};
+
+export const selectFieldsToExport = (): ExportThunkAction<
+  Promise<void>,
+  | SelectFieldsToExportAction
+  | FetchFieldsToExportAction
+  | FetchFieldsToExportErrorAction
+  | FetchFieldsToExportSuccessAction
+> => {
+  return async (dispatch, getState) => {
+    dispatch({
+      type: ExportActionTypes.SelectFieldsToExport,
+    });
+
+    const fieldsToExportAbortController = new AbortController();
+
+    dispatch({
+      type: ExportActionTypes.FetchFieldsToExport,
+      fieldsToExportAbortController,
+    });
+
+    const {
+      export: { query, namespace },
+      dataService: { dataService },
+    } = getState();
+
+    try {
+      const gatherFieldsResult = await gatherFieldsFromQuery({
+        ns: namespace,
+        abortSignal: fieldsToExportAbortController.signal,
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        dataService: dataService!,
+        query,
+        sampleSize: 50,
+      });
+
+      if (
+        fieldsToExportAbortController.signal.aborted ||
+        gatherFieldsResult.aborted
+      ) {
+        dispatch({
+          type: ExportActionTypes.FetchFieldsToExportError,
+          aborted: true,
+        });
+        return;
+      }
+
+      const fields: FieldsToExport = {};
+      for (const schemaPath of gatherFieldsResult.paths) {
+        fields[getIdForSchemaPath(schemaPath)] = {
+          path: schemaPath,
+          // We start all of the fields as unchecked.
+          selected: false,
+        };
+      }
+
+      dispatch({
+        type: ExportActionTypes.FetchFieldsToExportSuccess,
+        fieldsToExport: fields,
+      });
+    } catch (err: any) {
+      if (fieldsToExportAbortController.signal.aborted) {
```

Comment:
How would you get here given that aborted wouldn't throw and you'd land in the gatherFieldsResult.aborted check above?

I also don't think that aborted is an error here. Right now you can't really abort it, so I wouldn't worry about it and in future aborting would just mean "skip", ie. it still succeeds and the fact that it aborted is really just metadata. ie. you could use it to decide if you should display a banner.

I'd also only try {} catch {} around the `await gatherFieldsFromQuery()` line. Errors outside of that are programmer errors and we want those to propagate, not "runtime" errors like network or database ones.

Reply:
Makes sense, thanks for catching. Removed that check in the catch and moved the other parts out of the try catch. 👍